lints/lint_subject_contains_malformed_arpa_ip.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+// arpaMalformedIP is a linter that warns for malformed names under the
+// .in-addr.arpa or .ip6.arpa zones.
+// See also: lint_subject_contains_reserved_arpa_ip.go for a lint that ensures
+// well formed rDNS names in these zones do not specify an address in a IANA
+// reserved network.
+type arpaMalformedIP struct{}
+
+// Initialize for an arpaMalformedIP linter is a NOP to statisfy linting
+// interfaces.
+func (l *arpaMalformedIP) Initialize() error {
+	return nil
+}
+
+// CheckApplies returns true if the certificate contains any names that end in
+// one of the two designated zones for reverse DNS: in-addr.arpa or ipv6.arpa.
+func (l *arpaMalformedIP) CheckApplies(c *x509.Certificate) bool {
+	names := append([]string{c.Subject.CommonName}, c.DNSNames...)
+	for _, name := range names {
+		name = strings.ToLower(name)
+		if strings.HasSuffix(name, rdnsIPv4Suffix) ||
+			strings.HasSuffix(name, rdnsIPv6Suffix) {
+			return true
+		}
+	}
+	return false
+}
+
+// Execute will check the given certificate to ensure that all of the DNS
+// subject alternate names that specify a reverse DNS name under the respective
+// IPv4 or IPv6 arpa zones are well formed. A Warn LintResult is returned if
+// the name is in a reverse DNS zone but has the wrong number of labels.
+func (l *arpaMalformedIP) Execute(c *x509.Certificate) *LintResult {
+	for _, name := range c.DNSNames {
+		name = strings.ToLower(name)
+		var err error
+		if strings.HasSuffix(name, rdnsIPv4Suffix) {
+			// If the name has the in-addr.arpa suffix then it should be an IPv4 reverse
+			// DNS name.
+			err = lintReversedIPAddressLabels(name, false)
+		} else if strings.HasSuffix(name, rdnsIPv6Suffix) {
+			// If the name has the ipv6.arpa suffix then it should be an IPv6 reverse
```

Comment:
Comment should reference ip6.arpa. There's no **v** in the zone name. See [IETF](https://tools.ietf.org/id/draft-ietf-dnsop-isp-ip6rdns-04.html) .